alert-jira/src/main/java/com/synopsys/integration/alert/jira/common/util/JiraIssueHandler.java

```diff
@@ -120,64 +143,53 @@ protected boolean transitionIssue(IssueResponseModel issueModel, IssueConfig iss
-            if (responseContent.has("errors")) {
-                JsonObject errors = responseContent.get("errors").getAsJsonObject();
-                if (errors.has("reporter")) {
-                    JsonElement reporterErrorMessage = errors.get("reporter");
-                    if (null != reporterErrorMessage) {
-                        return new AlertFieldException(Map.of(
-                            getIssueCreatorFieldKey(),
-                            String.format("There was a problem assigning '%s' to the issue. Please ensure that the user is assigned to the project and has permission to transition issues. Error: %s", issueCreatorEmail, reporterErrorMessage)
-                        ));
-                    }
-                } else {
-                    Set<Map.Entry<String, JsonElement>> entries = errors.entrySet();
-                    List<String> fieldErrors = entries.stream()
-                                                   .map(entry -> String.format("Field '%s' has error %s", entry.getKey(), entry.getValue()))
-                                                   .collect(Collectors.toList());
-                    responseErrors.addAll(fieldErrors);
-                }
-
-            }
-            if (responseContent.has("errorMessages")) {
-                JsonArray errorMessages = responseContent.get("errorMessages").getAsJsonArray();
-                for (JsonElement errorMessage : errorMessages) {
-                    responseErrors.add(errorMessage.getAsString());
+            try {
+                List<String> responseErrors = extractErrorsFromResponseContent(responseContent, issueCreatorEmail);
+                if (!responseErrors.isEmpty()) {
+                    message += " | Details: " + StringUtils.join(responseErrors, ", ");
                 }
+            } catch (AlertFieldException reporterException) {
+                return reporterException;
             }
         }
-
-        String message = restException.getMessage();
-        if (!responseErrors.isEmpty()) {
-            message += " | Details: " + StringUtils.join(responseErrors, ", ");
-        }
-
         return new AlertException(message, restException);
     }
 
-    private void addIssueProperties(String issueKey, JiraIssueSearchProperties issueProperties) throws IntegrationException {
-        jiraIssuePropertyHelper.addPropertiesToIssue(issueKey, issueProperties);
-    }
-
-    private IssueRequestModelFieldsBuilder createFieldsBuilder(IssueContentModel contentModel) {
-        IssueRequestModelFieldsBuilder fieldsBuilder = new IssueRequestModelFieldsBuilder();
-        fieldsBuilder.setSummary(contentModel.getTitle());
-        fieldsBuilder.setDescription(contentModel.getDescription());
+    private List<String> extractErrorsFromResponseContent(JsonObject responseContent, String issueCreatorEmail) throws AlertFieldException {
+        List<String> responseErrors = new ArrayList<>();
+        if (responseContent.has("errors")) {
```

Comment:
We could also use Optional's to reduce the cognitive complexity here. Something like this
``` java
Optional<JsonObject> errorsOptional = Optional.ofNullable(responseContent).filter(json -> json.has("errors")).map(json -> json.get("errors").getAsJsonObject());
Optional<JsonElement> errorsReporterOptional = errorsOptional.filter(errorsJson -> errorsJson.has("reporter")).map(errorsJson -> errorsJson.get("reporter"));
if (errorsReporterOptional.isPresent()) {
    throwReporterException(errorsReporterOptional.get(), issueCreatorEmail);
}
```

The optionals should help reduce the nested if's when dealing with the JsonObjects.

Reply:
Hmm, I've always had reservations about using inline `Optional.ofNullable(...)`, but other than a temporary object that will be garbage collected, this is definitely easier to follow.